internal/services/databricks/databricks_access_connector_resource.go

```diff
@@ -42,7 +44,46 @@ func (r AccessConnectorResource) Arguments() map[string]*pluginsdk.Schema {
 
 		"resource_group_name": commonschema.ResourceGroupName(),
 
-		"identity": commonschema.SystemAssignedIdentityOptional(),
+		// cannot use common schema 'SystemAssignedUserAssignedIdentityOptional' because
+		// the Databricks implementation is slightly different as it only allows
+		// 'SystemAssigned' or 'UserAssigned' (e.g. 'Only SystemAssigned or
+		// UserAssigned Identity is supported for an Access Connector
+		// resource, not both together.') and only allows for a single 'identity_ids'
+		// to be passed...
+		"identity": {
```

Comment:
rather than doing this, let's introduce a new `commonschema` type to support this, `commonschema.SystemAssignedOrUserAssignedIdentity` - `identity` is intentionally standardised across the provider now